department-of-veterans-affairs/abd-vro
# Authorize Generic user for Streamlit

## What was the problem?

Currently, anyone with access to the deployed url via CAG would be able access the application. There is no authentication or authorization protecting any part of the application.

Associated tickets or Slack threads:
- #3555

## How does this fix it?[^1]

This adds a level of authentication via github OAuth apps. Anyone with a github account will be able to access the application as a basic user. Eventually more authorization will be added to only allow people who are part of the department-of-veterans-affairs organization and part of specific teams. 

## How to test this PR
1. Pull branch
2. from root dir, run `./gradlew vro-streamlit:check` to run all checks/lints/unit tests.
3. from root dir, run `./gradlew :vro-streamlit:docker ` then run `COMPOSE_PROFILES="streamlit" ./gradlew :dockerComposeUp` to run the container. Then open browser to http://0.0.0.0:8501/ to view webpages. 
4. On the `Home` page, click the login with github and authorize to view the BIE Events pages. 

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
